### PR TITLE
Bookstore: add reducers and actions

### PR DESCRIPTION
**Pull request by mirouhml**
Write book's actions and reducer (in `/src/redux/books/books.js`) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).

- Define action types for adding and removing a book.
- Set the initial state to be an empty array of books.
- Export Action Creators for the actions.
- Write a reducer and export it as default.
  - Define state changes for the actions created.
  - In case of unknown action - return the current state.

Write categories actions and reducer (in `/src/redux/categories/categories.js`) using the [ducks pattern](https://github.com/erikras/ducks-modular-redux).

- Define an action type for displaying "under construction" message.
- Set the initial state to be an empty array of categories.
- Export Action Creators for the actions.
- Write a reducer and export it as default.
  - For the check status action return a string "Under construction".
  - In case of unknown action - return the current state.

Configure the Redux Store (`/src/redux/configureStore.js`):

- Importing the necessary methods from Redux.
- Combine both reducers into a root reducer by using combineReducers function.
- Use the root reducer to create the store.

### CODE REVIEW

File: src/redux/configureStore.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
...
+import * as categoryReducers from './categories/categories';
+
+const rootReducer = combineReducers({
+  books: bookReducers,
+  categories: categoryReducers,
+});
+
+const store = createStore(rootReducer);
+
+export default store;
```
Comment (GzimAsani) [inline src/redux/configureStore.js]: - [ ] Please check this console error and please try to fix it
Screenshot from 2022-03-28 19-30-37

**Review by GzimAsani — CHANGES_REQUESTED** [review]
# Hi  @mirouhml     
Good job so far! You are almost done :+1:  :+1: 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To highlight:
- Good job with the git-flow :+1:
- Good PR description :clap: 

## Required Changes ♻️ 
- [ ] _Check the comments below :arrow_double_down: _

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
N/A

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/redux/configureStore.js (hunk 1 above)
Comment (mirouhml) [reply src/redux/configureStore.js]: @GzimAsani thanks for catching that I fixed it!

**Review by GzimAsani — APPROVED** [review]
# STATUS: APPROVED :100:         
## Thank you for implementing all the project requirements, you've done a great job.  :trophy: There is nothing else to say other than... it's time to merge it :champagne:  :tada: Congratulations

Cheers and Happy coding! :clap:  :computer:  :wine_glass: 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Please update your Readme file accordingly
 
 _As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s).

**Comment by mirouhml** [comment]
Thanks for the review @GzimAsani !

**Pull request merged**